src/plugins/cache/cache.c

```diff
@@ -151,7 +151,7 @@ static char * elektraGenTempFilename (char * cacheFileName)
 
 	tmpFilenameSize = strlen (cacheFileName) + POSTFIX_SIZE;
 	tmpFile = elektraCalloc (tmpFilenameSize);
-	len = sprintf (tmpFile, "%s", cacheFileName);
+	len = snprintf (tmpFile, strlen (cacheFileName), "%s", cacheFileName);
```

Comment:
As far as I can tell, this line contains an off-by-one error. According to the documentation of `strlen` the function

> returns the number of characters that precede the terminating NUL character

. The function `snprintf` on the other hand

> will write at most size-1 of the characters printed into the output string (the size'th character then gets the terminating `\0')

. If we assume `cacheFileName` is `"one"`, then `strlen` returns `3`, while `snprintf` will write two characters and `'\0'`, effectively storing `"on"` in `tmpFile`.